spec/lib/iiif_manifest/v3/manifest_builder/canvas_builder_spec.rb

```diff
@@ -12,151 +12,156 @@
+  let(:record) do
+    MyWork.new(display_content: display_content)
+  end
 
-    context 'when media_fragment is defined' do
-      before do
-        allow(record).to receive(:media_fragment).and_return('xywh=160,120,320,240')
-      end
-      it 'returns a canvas url' do
-        expect(builder.path).to eq 'http://test.host/books/book-77/manifest/canvas/test-22#xywh=160,120,320,240'
+  before do
+    class MyWork
+      attr_reader :display_content
+
+      def initialize(display_content:)
+        @display_content = display_content
       end
 
-      context 'and is blank' do
-        before do
-          allow(record).to receive(:media_fragment).and_return(nil)
-        end
-        it 'returns a canvas url' do
-          expect(builder.path).to eq 'http://test.host/books/book-77/manifest/canvas/test-22'
-        end
+      def id
+        'test-22'
       end
     end
+
+    allow(body_builder).to receive(:apply).and_return(iiif_body)
+    allow(body_builder_factory).to receive(:new).and_return(body_builder)
+    allow(iiif_annotation_factory).to receive(:new).and_return(iiif_annotation)
+    allow(content_builder).to receive(:new).and_return(built_content)
+    allow(thumbnail_builder).to receive(:apply).and_return(iiif_thumbnail)
+    allow(thumbnail_builder_factory).to receive(:new).and_return(thumbnail_builder)
   end
 
-  describe '#canvas' do
-    let(:record) do
-      MyWork.new
-    end
+  let(:iiif_body) do
+    body = IIIFManifest::V3::ManifestBuilder::IIIFManifest::Body.new
+    body['width'] = '100px'
+    body['height'] = '100px'
+    body['duration'] = nil
+    body
+  end
 
-    after do
-      Object.send(:remove_const, :MyWork)
-    end
+  let(:iiif_thumbnail) do
+    thumbnail = IIIFManifest::V3::ManifestBuilder::IIIFManifest::Thumbnail.new
+    thumbnail['width'] = 200
+    thumbnail['height'] = 150
+    thumbnail['duration'] = nil
+    thumbnail
+  end
 
-    context 'when the display content is empty for an item' do
-      before do
-        class MyWork
-          def id
-            'test-22'
-          end
+  let(:iiif_annotation) do
+    annotation = IIIFManifest::V3::ManifestBuilder::IIIFManifest::Annotation.new
+    annotation.body = iiif_body
+    annotation
+  end
 
-          def display_content
-            []
-          end
-        end
-      end
+  let(:iiif_annotation_factory) do
+    double
+  end
+
+  let(:body_builder) do
+    instance_double(IIIFManifest::V3::ManifestBuilder::BodyBuilder)
+  end
+
+  let(:body_builder_factory) do
+    double
+  end
+
+  let(:thumbnail_builder) do
+    instance_double(IIIFManifest::V3::ManifestBuilder::ThumbnailBuilder)
+  end
+
+  let(:thumbnail_builder_factory) do
+    double
+  end
+
+  let(:built_content) do
+    IIIFManifest::V3::ManifestBuilder::ContentBuilder.new(
+      record.display_content,
+      iiif_annotation_factory: iiif_annotation_factory,
+      body_builder_factory: body_builder_factory
+    )
+  end
 
+  let(:content_builder) do
+    double
```

Comment:
```suggestion
    double("Content Builder")
```

Give the double a name.  If there's an error / failing spec and if it's related to this double the name will be part of the message.